### ISSUE TITLE
Blocking of content source websites not working in 3.10.4

### ISSUE THREAD

**Issue by dhowe** [dhowe/AdNauseam#1847]
[submitted via email]
> The issue with the uBlock interface for directly blocking content source websites is still not working in the new version — in fact, it seems to have become worse.
> 
> Not only is there no ability to remove websites from being blocked (change a site set to "Red" on the graphical interface to "Green", or even "Grey"), but when I do try to make changes, the "activate" button doesn't even appear to let me see what the result of changing the blocking status of a site; just the "Make Permanent" (lock icon) and "Revert" (eraser icon) buttons appear, and seem to do nothing at all when pressed.

Win 10 Home
Firefox 88.0.1 (64bit)
AdNauseam 3.10.4

**Comment by mneunomne** [comment]
@dhowe The block itself seems to be working, the problem is that the classes aren't being added when the user clicked on the boxes, and thus they don't remain with the colors as the should.
In Ublock for instance after the user clickes on the box this happens (even after you re-open the interface, reloads the page etc):
![Screen Shot 2021-05-22 at 13 18 34](https://user-images.githubusercontent.com/4967860/119224747-3cd63f00-bb00-11eb-943a-f8612566e854.png)
With the following html class:
```html
<span data-src="." data-acount="0" data-bcount="0" class="blockRule ownRule"> </span>
```

But in Adn always stays grey with the class `undefinedRule`, making the user not knowing if the rule was added or not:
![Screen Shot 2021-05-22 at 13 19 32](https://user-images.githubusercontent.com/4967860/119224779-5ecfc180-bb00-11eb-884c-3f6eff86e601.png)
At least the rules is being correctly added in **My Rules**:
![Screen Shot 2021-05-22 at 13 20 31](https://user-images.githubusercontent.com/4967860/119224859-c259ef00-bb00-11eb-9b89-726eabe8d897.png)

**Comment by mneunomne** [comment]
Found where the issue is. It is with the lookupRuleData function, which ends up causing the `rule.action` not to be understood further in the code when setting up the classes. The RuleData should be an Object, but it is coming as a string.

Our overwritten method is quite different from the original one. 

Ours:
https://github.com/dhowe/AdNauseam/blob/a61a94dae04590c6822d340cb036cc626520a11a/src/js/dynamic-net-filtering.js#L374-L380

Original one:
https://github.com/dhowe/AdNauseam/blob/a61a94dae04590c6822d340cb036cc626520a11a/src/js/dynamic-net-filtering-orig.js#L394-L405

**Comment by dhowe** [comment]
this is the code from ub 1.33:



      
      
        Matrix.prototype.lookupRuleData = function(src, des, type) {
            var r = this.evaluateCellZY(src, des, type);
            if ( r === 0 ) {
                return null;
            }
            return {
                src: this.z,
                des: this.y,
                type: this.type,
                action: r === 1 ? 'block' : (r === 2 ? 'allow' : 'noop')
            };
        };

**Comment by mneunomne** [comment]
@dhowe I realized that for some reason we are using the https://github.com/gorhill/uBlock/blob/1.35.2/src/js/dynamic-net-filtering.js later than 1.33.0 version. Maybe we were merging we got the wrong version or something... I think the best solution is to sync Adn to uBlock 1.35.2 because there was a big refactor in the logger system.

**Comment by dhowe** [comment]
sounds good to me

**Comment by mneunomne** [comment]
@dhowe Just merged uBlock1.34.0 into Adnauseam and this seems to be fixed. Maybe we can do a quick release to fix this? But since it's a merge there is quite a lot of testing to be made once again.

https://github.com/mneunomne/AdNauseam/tree/merge1.34.0

**Comment by mneunomne** [comment]
And btw we need to add to the test set these functions of clear and lock in the dynamic filtering UI. They are not working properly in our currrent version as the user in the email suggested. Now its working on 1.34.0

![Screen Shot 2021-06-01 at 12 38 03](https://user-images.githubusercontent.com/4967860/120310086-46b52a80-c2d6-11eb-8977-665dac24c7f9.png)
![Screen Shot 2021-06-01 at 12 38 12](https://user-images.githubusercontent.com/4967860/120310081-4452d080-c2d6-11eb-9dae-783275bbe43c.png)
![Screen Shot 2021-06-01 at 12 38 07](https://user-images.githubusercontent.com/4967860/120310085-4583fd80-c2d6-11eb-8cc8-18a0cd1c1fdb.png)

**Comment by mneunomne** [comment]
This is fixed with merge 1.34.0. Closing
